### PR TITLE
Simplify gitignore for mu-plugins

### PR DESCRIPTION
**Pull request by aaemnnosttv**
In my experience, I haven't had a situation where I had a mu-plugin in the root of the mu-plugins directory that I would want to ignore as these are almost always self-created.  Sub directories, however, are usually ok to ignore as a rule, as those can be packages, which should be ignored.  Since packages can't be installed into the root, without some kind of custom installer, it makes more sense I think to have it like this.  Of course there could be sub-directories there that you may not want to ignore, and then in those situations, it makes perfect sense to add an exception in the ignore.

From a workflow perspective, it seems unintuitive to me that if I want to add an mu-plugin of my own in the root, that I should then need to add it to the gitignore whitelist.

**TL;DR**
This changes the `.gitignore` for the mu-plugins directory from ignoring everything below `/*` and then explicitly un-ignoring `!` specific mu-plugin files, to just ignoring sub-directories instead `/*/`.

### CODE REVIEW

**Comment by swalkinshaw** [comment]
I think this does make more sense. The entire point of these git ignore rules is to ignore directories that are managed by package managers (Composer). And you're right that the root of `mu-plugins/` doesn't fit that criteria.

**Comment by retlehs** [comment]
it's wrong to assume that everything in a subfolder in the mu-plugins folder would be there from a package manager. 

why not just remove the mu-plugins directory gitignore rules altogether?

**Comment by aaemnnosttv** [comment]
I'm not implying that any sub-directory there would be from a package manager, but in this case, `wordpress-muplugin` is a package type, with `web/app/mu-plugins/{$name}/` as the installer path.  

It doesn't make sense that all other package installer paths be ignored while this one is not.

**Comment by retlehs** [comment]
we don't have a gitignore for themes. we should just remove the gitignore for mu-plugins.

**Comment by retlehs** [comment]
ignore me :tired_face:

**Comment by retlehs** [comment]
thanks!

**Comment by QWp6t** [comment]
Couldn't this same reasoning be used to change the gitignore rule for the plugins directory as well?
